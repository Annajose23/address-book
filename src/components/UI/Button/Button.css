.button {
    border: 1px solid #eee;
    text-align: center;
    background-color: #3A3B3C;
    color:white;
    padding: 16px;
    border-radius: 5px;
}