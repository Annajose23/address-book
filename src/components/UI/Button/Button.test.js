import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Button from "./Button";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

describe("Button", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Button clicked={jest.fn()} />);
  });

  it("should render button component with props", () => {
    expect(wrapper).not.toBeNull();
  });

  it("should contain button tag", () => {
    expect(wrapper.find("button")).toHaveLength(1);
  });

  it("should contain button tag with classname button", () => {
    expect(wrapper.find("button").hasClass("button")).toEqual(true);
  });

  it("render correctly button component", () => {
    const ButtonComponent = renderer
      .create(<Button clicked={jest.fn()} />)
      .toJSON();
    expect(ButtonComponent).toMatchSnapshot();
  });
});
