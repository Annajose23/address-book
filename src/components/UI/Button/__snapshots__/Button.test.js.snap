// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render correctly button component 1`] = `
<button
  className="button"
  onClick={[MockFunction]}
/>
`;
